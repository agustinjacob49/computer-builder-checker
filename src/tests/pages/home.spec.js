/* eslint-disable testing-library/no-unnecessary-act */
/* eslint-disable testing-library/prefer-screen-queries */
import { render, fireEvent, screen, act } from '@testing-library/react';
import View from './../../pages/home/view.js';

describe('Home view', () => {
  test('Snapshot testing', () => {
    const { container } =  render(<View />);
    expect(container).toMatchSnapshot();
  });


  test('Click on render button when input is empty should show errorMessage', () => {
    render(<View />);
    const button = screen.getByTestId('button-render');

    fireEvent.click(button);

    screen.getByText('Please, enter some text', { exact: false });
  });

  test('Click on clean should clean text area', () => {
    render(<View />);
    const textarea = screen.getByRole("textbox");
    fireEvent.input(textarea, { target: { value: "A DEPENDS B C" } });

    expect(textarea.value).toBe("A DEPENDS B C");

    const buttonClean = screen.getByTestId('button-clean');
    fireEvent.click(buttonClean);
    
    expect(textarea.value).toBe("");
  });

  jest.setTimeout(10000);
  test('Enter circular reference', async () => {
    render(<View />);
    const button = screen.getByTestId('button-render');

    const textarea = screen.getByRole("textbox");

    act(() => {
      fireEvent.input(textarea, { target: { value: "A DEPENDS B C\nC DEPENDS X\nX DEPENDS A" } });
    });

    act(() => {
      fireEvent.click(button);
    })

    await screen.findByText('Circular reference', { exact: false });
   
  });

  jest.setTimeout(15000);
  test('Enter text should render a graph', async () => {
    render(<View />);
    const button = screen.getByTestId('button-render');

    const textarea = screen.getByRole("textbox");

    act(() => {
      fireEvent.input(textarea, { target: { value: "A DEPENDS B C\nC DEPENDS X" } });
    });

    act(() => {
      fireEvent.click(button);
    })

    await screen.findAllByTestId("vertex");
  });

});

