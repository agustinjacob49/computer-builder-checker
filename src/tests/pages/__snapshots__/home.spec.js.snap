// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home view Snapshot testing 1`] = `
<div>
  <div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <div>
          <div
            class="form-floating"
          >
            <textarea
              class="input-text"
              data-testid="input-text-area"
              id="floatingTextarea"
            />
          </div>
        </div>
      </div>
      <div
        class="col"
      >
        <fieldset />
      </div>
    </div>
    <div
      class="button-group"
    >
      <button
        class="btn btn-primary"
        data-testid="button-render"
        type="button"
      >
        Render
      </button>
       
      <button
        class="btn btn-secondary"
        data-testid="button-clean"
        type="button"
      >
        Clean
      </button>
       
    </div>
  </div>
</div>
`;
